taskchampion/xtask/src/main.rs

```diff
@@ -29,3 +41,106 @@ fn codegen() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// `cargo xtask msrv (X.Y)`
+///
+/// This checks and updates the Minimum Supported Rust Version for all files specified in the `MSRV_FILE_PATHS` const in `xtask/src/main.rs` where the pattern `MSRV = "X.Y"` is found in the file.
+/// The argument "X.Y" will replace the text any place the MSRV is found.
+fn msrv(args: Vec<String>) -> anyhow::Result<()> {
+    if args.len() != 2 {
+        // check that (X.Y) argument is (mostly) valid:
+        if !args[2].chars().all(|c| c.is_numeric() || c == '.') {
+            anyhow::bail!("xtask: Invalid argument format. Xtask MSRV argument takes the form \"X.Y(y)\", where XYy are numbers. eg: `cargo run xtask MSRV 1.68`");
+        }
+
+        // for each file in const paths vector
+        for path in MSRV_FILE_PATHS {
+            let path: &Path = Path::new(&path);
+            let mut is_pattern_in_file = false;
+
+            // check if path exists, if not, skip then continue
+            if !path.exists() {
+                println!(
+                    "xtask: Warning: xtask: Path \"{}\" not found. Skipping.",
+                    path.display()
+                );
+                continue;
+            }
+
+            let mut file: File = File::options().read(true).write(true).open(path)?;
+            let reader = BufReader::new(&file);
+
+            // set the file_extension if it exists and is valid UTF-8
+            let mut file_extension = String::new();
```

Comment:
The clippy lint here is pointing out that `file_extension` is overwritten before it is read in every codepath, so it doesn't need to be initialized here. Same thing below for `comment_glyph`.